src/reader.rs

```diff
@@ -19,12 +19,42 @@ enum PipedChar {
     EOF,
 }
 
+#[derive(Clone)]
```

Comment:
what is this for?

Reply:
I guess, it's necessary for `vec!` macros

https://github.com/philippkeller/rexpect/blob/0212cd6ff069e5633ea8e6d4dad8212f88ef6464/src/session.rs#L495

Reply:
Not sure about `vec!` probably I got mistaken but there's a real cause here. Currently `string_needle` method consumes/owns a `ReadUntil ` instance. I hope the convert methods could be changed to have a `&self` parameter rather than `self` in which case the `Clone` could be deleted.

https://github.com/philippkeller/rexpect/blob/0212cd6ff069e5633ea8e6d4dad8212f88ef6464/src/session.rs#L503-L508